test/integration/component/test_multiple_nic_support.py

```diff
@@ -224,7 +224,7 @@ def setUpClass(cls):
                     cls.virtual_machine1.default_network_id = nic.networkid
                     break
         except Exception as e:
-            cls.fail("Exception while deploying virtual machine: %s" % e)
+            cls.fail(f"Exception while deploying virtual machine: {e}")
```

Comment:
@DaanHoogland 
a bit strange that this does not work in centos7/python2

```
======================================================================
ERROR: Failure: SyntaxError (invalid syntax (test_multiple_nic_support.py, line 227))
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 132, in run
    self.beforeTest(result)
  File "/usr/lib/python2.7/site-packages/nose/case.py", line 74, in beforeTest
    beforeTest(self.test)
  File "/usr/lib/python2.7/site-packages/nose/proxy.py", line 117, in beforeTest
    self.plugins.beforeTest(self.test)
  File "/usr/lib/python2.7/site-packages/nose/plugins/manager.py", line 99, in __call__
    return self.call(*arg, **kw)
  File "/usr/lib/python2.7/site-packages/nose/plugins/manager.py", line 167, in simple
    result = meth(*arg, **kw)
  File "/usr/lib/python2.7/site-packages/marvin/marvinPlugin.py", line 174, in beforeTest
    join([self.__identifier, self.__testName])
TypeError: sequence item 0: expected string, NoneType found
-------------------- >> begin captured stdout << ---------------------
=== TestName: Failure: | Status : EXCEPTION ===
```

Reply:
not sure why, but the old way works (and still does in py3) no worries. I'll give it a look one of these days.

Reply:
ok, double checked:
```
>>> e = Exception("root cause")
>>> raise Exception(f"nested exception {e}")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
Exception: nested exception root cause
```
I'll fix if it turns out to be a problem for anyone ;)